lib/__tests__/cloudworker-e2e.test.js

```diff
@@ -211,4 +211,186 @@ describe('cloudworker-e2e', async () => {
     await upstream.close()
     cb()
   })
+
+  test('simple sha256 encode', async (cb) => {
+    const script = `
+      addEventListener('fetch', event => {
+        const url = new URL(event.request.url)
+        event.respondWith(respondWithHelloAsSHA256())
+      })
+
+      function byteStringToUint8Array(byteString) {
+        const ui = new Uint8Array(byteString.length)
+        for (let i = 0; i < byteString.length; ++i) {
+          ui[i] = byteString.charCodeAt(i)
+        }
+        return ui
+      }
+
+      function toHexString(buffer) {
+        var s = '', h = '0123456789abcdef';
+        (new Uint8Array(buffer)).forEach((v) => { s += h[v >> 4] + h[v & 15]; })
+        return s;
+      }
+
+      async function respondWithHelloAsSHA256() {
+        const sha = await toSHA256("hello")
+        return new Response(sha, {status: 200})
+      }
+
+      async function toSHA256(message) {
+        let hash = await crypto.subtle.digest('SHA-256', byteStringToUint8Array(message))
+        return toHexString(hash)
+      }
+    `
+    const server = new Cloudworker(script).listen(8080)
+    const res = await axios.get('http://localhost:8080', defaultAxiosOpts)
+    expect(res.status).toEqual(200)
+    // taken from `echo -n "hello" | shasum -a 256`
+    expect(res.data).toContain('2cf24dba5fb0a30e26e83b2ac5b9e29e1b161e5c1fa7425e73043362938b9824')
+    await server.close()
+    cb()
+  })
+
+  test('simple hmac sign', async (cb) => {
+    const script = `
+      addEventListener('fetch', event => {
```

Comment:
For long e2e scripts like this, please add as a fixture file instead. It helps with readability.